ldap/servers/slapd/back-ldbm/db-bdb/bdb_layer.c

```diff
@@ -1778,6 +1778,112 @@ bdb_instance_start(backend *be, int mode)
     return return_value;
 }
 
+static int
+bdb_get_page_count(dbi_db_t *db, dbi_txn_t *txn, uint32_t *count)
+{
+    DB_BTREE_STAT *stats = NULL;
+    int rc;
+
+    rc = ((DB*)db)->stat(db, (DB_TXN*)txn, (void *)&stats, 0);
+    if (rc != 0) {
+        slapi_log_err(SLAPI_LOG_ERR, "bdb_get_page_count",
+                      "Failed to get bd statistics: db error - %d %s\n",
+                      rc, db_strerror(rc));
+        rc = DBI_RC_OTHER;
+    }
+    *count = rc ? 0 : stats->bt_pagecnt;
+    slapi_ch_free((void **)&stats);
+    return rc;
+}
+
+/*
+ * Get the page count for this backend instance
+ * If any error occurs just return 0
+ */
+uint32_t
+bdb_get_inst_page_count(struct ldbminfo *li, ldbm_instance *inst)
+{
+    if (!inst->inst_id2entry) {
```

Comment:
Sorry I missed where it is set in bdb code? It am under the impression it is always NULL in bdb.

Reply:
What are you speaking about ? inst ? 
 if that is the case, you are mistaken: probably doing a confusion between:
       ldbm  ( the back-ldbm code for backend backed on databases)
       lmdb : the lmdb database
 injst is the be->be_private data containing ldbm context and is never NULL

Reply:
It's not about inst, its about "inst_id2entry".  When a backend is not initialized this is NULL so we must check for this  to properly handle the count.  This is not a mistake.